byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -105,6 +152,95 @@ instance Embed BBODY CHAIN where
 instance Embed PBFT CHAIN where
   wrapFailed = PBFTFailure
 
+instance Embed (UTXOWS TxId) CHAIN where
+  wrapFailed = LedgerUTXOFailure
+
+instance Embed DELEG CHAIN where
+  wrapFailed = LedgerDelegationFailure
+
 genesisHash :: Hash
 -- Not sure we need a concrete hash in the specs ...
 genesisHash = Crypto.Hash.hash ("" :: ByteString)
+
+ -- | Lens for the delegation interface state contained in the chain state.
+disL :: Lens' (State CHAIN) DIState
+disL = _5
+
+instance HasTrace CHAIN where
+
+  initEnvGen = (,,,) <$> gCurrentSlot <*> pure initialUTxO <*> initEnvGen @DELEG <*> gPParams
+    where
+      -- If we want to generate large traces, we need to set up the value of the
+      -- current slot to a sufficiently large value.
+      gCurrentSlot = Slot <$> Gen.integral (Range.constant 32768 2147483648)
+
+      -- TODO: For now we generate an empty UTxO. In later iterations we will
+      -- incorporate the UTxO payload into the chain.
+      initialUTxO = UTxO Map.empty
+
+      gPParams
+        = PParams
+        <$> gMaxHeaderSize
+        <*> gMaxBlockSize
+        <*> gMaxTxSize
+        <*> gMaxProposalSize
+        <*> gBlockSigCntThreshold
+        <*> gSlotsPerEpoch
+        <*> gUpdateProposalTTL
+        <*> pure 1 -- _scriptVersion
+        <*> gConfirmationTreshold
+        <*> gAdoptinThreshold
+        <*> pure k -- _stableAfter
+        <*> pure 0 -- factor a
+        <*> pure 0 -- factor b
+        where
+          -- In mainet the maximum header size is set to 2000000 and the maximum
+          -- block size is also set to 2000000, so we have to make sure we cover
+          -- those values here. The upper bound is arbitrary though.
+          gMaxHeaderSize = Gen.integral (Range.constant 0 4000000)
+          gMaxBlockSize  = Gen.integral (Range.constant 0 4000000)
+
+          gMaxTxSize = Gen.integral (Range.constant 0 4000000)
+          gMaxProposalSize = Gen.integral (Range.constant 0 4000000)
+
+          gBlockSigCntThreshold = pure (1/5) -- TODO: this needs to be aligned with the formal specs.
+
+          -- The number of slots per epoch is computed from 'k':
+          -- slots per-epoch = k * 10
+          gSlotsPerEpoch = pure $! SlotCount $ unBlockCount k *  10
+
+          gUpdateProposalTTL = SlotCount <$> Gen.integral (Range.linear 1 100)
+
+          -- Confirmation threshold
+          gConfirmationTreshold = Gen.integral (Range.linear 1 7)
+
+          -- Update adoption threshold
+          gAdoptinThreshold = Gen.integral (Range.linear 1 7)
+
+  sigGen (_sNow, _utxo0, dsEnv, _pps) (Slot s, _sgs, h, _utxo, ds, _us) = do
+    vkI <- Gen.element $ Bimap.elems (ds ^. dms)
```

Comment:
If we don't care about which element we pick we can use `dontShrink`.

Reply:
Specify why we don't need to shrink ...

Reply:
How about pulling generators like `gPParams` outside the class instance? They'd be useful for other STSes too.

Reply:
A *proper* generator for protocol parameters should be in a module like `Ledger.Update.Generators` or something like that sure. However in this case, this is a temporary generator that I use till we align the formal and executable specs (see TODO above for instance).